src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -329,27 +329,36 @@ object JoinIndexRule extends Rule[LogicalPlan] with Logging {
       plan: LogicalPlan,
       allIndexes: Seq[IndexLogEntry]): Seq[IndexLogEntry] = {
 
-    /* map of signature provider to signature for this subplan */
-    val signatureMap: mutable.Map[String, String] = mutable.Map()
-
-    def signatureValid(entry: IndexLogEntry): Boolean = {
-      val sourcePlanSignatures = entry.source.plan.properties.fingerprint.properties.signatures
-      assert(sourcePlanSignatures.length == 1)
-      val sourcePlanSignature = sourcePlanSignatures.head
-
-      if (!signatureMap.contains(sourcePlanSignature.provider)) {
-        val signature: String = LogicalPlanSignatureProvider
-          .create(sourcePlanSignature.provider)
-          .signature(plan)
-        signatureMap.put(sourcePlanSignature.provider, signature)
-      }
+    RulesHelper.getLogicalRelation(plan) match {
+      case Some(r) =>
```

Comment:
same comment as in FilterIndexRule, copy pasting here.

Could you please explain the reason for this change? To me, this looks like hard-coding signature calculation logic to a signature provider which works only on LogicalRelation. I think here we should keep it relaxed to any "plan" node and not just logical relation. I also don't think it saves on calculation so some explanation could help. Thanks.

Reply:
I added an explanation to the previous comment in FilterIndex rule. Please check that and let me know if further discussion is required. Thnx